### ISSUE TITLE
Change default apache logrotate config

### ISSUE THREAD

**Issue by dolanjs** [freedomofpress/securedrop#304]
Change default apache logrotate to rotate every day and only retain 1 day. This should give enough time for any personnel to get on site for organizations that may opt not to enable ssh.
Since sensitive information could be in error messages the logs should be securely deleted.

**Comment by garrettr** [comment]
What is the current logrotate interval? We only store Apache logs for the Document Interface anyway, so this is not relevant to minimizing metadata about source behavior. If anything, given our current turnaround on support tickets, 1 day of retention is insufficient and would hamstring us in our efforts to do remote troubleshooting using the logs.

I propose we move this off the 0.4 milestone and revisit at a later date, or close this entirely.

**Comment by heartsucker** [comment]
Wouldn't this leak how often a journalist replies and downloads material
which would leak source behavior?

Am 15.03.2017 7:06 nachm. schrieb "Garrett Robinson" <
notifications@github.com>:

What is the current logrotate interval? We only store Apache logs for the
Document Interface anyway, so this is not relevant to minimizing metadata
about source behavior. If anything, given our current turnaround on support
tickets, 1 day of retention is insufficient and would hamstring us in our
efforts to do remote troubleshooting using the logs.

I propose we move this off the 0.4 milestone and revisit at a later date,
or close this entirely.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/freedomofpress/securedrop/issues/304#issuecomment-286830205>,
or mute the thread

**Comment by conorsch** [comment]
@garrettr The apache logs are currently set to 52 weekly rotations, meaning compressed logs will remain on disk for approximately one year. Since we don't log the Source Interface, that logging only applies to the Journalist Interface—however, the Journalist Interface logging is currently broken (#1606), so right now zero useful information exists on-disk about Journalist behavior.

**Comment by fowlslegs** [comment]
I think persisting logs for 2 weeks would be a reasonable first step. Ideally, we'd be able to reduce this further, but that will require some combination of better support turnaround, and implementation of the opt-in log reporting we've discussed.

**Comment by heartsucker** [comment]
@redshiftzero Hackathon candidate.

**Comment by redshiftzero** [comment]
A day is certainly too short, but a month or two is reasonable.

**Comment by conorsch** [comment]
Agreed with @redshiftzero: let's set the new default logrotate value to 1-2 months. Once the change is made in the Ansible config logic, a playbook run (via `./securedrop-admin install`) will be required to enforce the change. Future new installs will receive the updated value automatically. 

As stated above, the logrotate functionality is only relevant for the Journalist Interface (previously called the "Document Interface"; see #1384), as the Source Interface does not log visitor interactions in production.

**Comment by zenmonkeykstop** [comment]
Keeping, the existing PR would need to be updated to cover existing instances without a playbook run, but the change is still valid.

**Comment by legoktm** [comment]
I believe the status quo has changed since this was last seriously discussed, as Debian switched the default from 52 weeks (which SecureDrop inherited back in 2017) to 2 weeks, which is what [psivesely suggested](https://github.com/freedomofpress/securedrop/issues/304#issuecomment-286840114).

So the current proposal to set it to 30 days is actually **increasing** log retention, not decreasing it. My initial impression is that I don't think there's a problem for us to solve anymore. I don't believe anyone has ever argued for 1 month in the context of needing that much retention at minimum, just that 52 weeks was too long, and 1 month was a reasonable place to shorten it to (which I'd be on board with).

But now that we're at 2 weeks, I'm okay with leaving it like that, keeping in mind that I can't think of a time in the past 3 years having the extra retention would've been useful, and to increase the retention, it'll require extra maintenance work for us to get the PR merged and keep the logrotate file up to date with Debian.

----

I'll put this on the agenda for the first team meeting next year for further discussion.

**Comment by legoktm** [comment]
We discussed this today in the team meeting and agreed that given the default change to 2 weeks, there's no needed action from us here and to close it as resolved.
